### PR TITLE
Tmp issue 186

### PR DESCRIPTION
**Pull request by diptighosalkar**
Set method done partly for sigbuilder block

It include implementation of set and other features in case where Launch graphic window(y/n)? is n  (No graph is invoke)

### CODE REVIEW

File: SendLog.py [hunk 1]
```diff
@@ -1724,6 +1726,110 @@ def run_scilab_func_expr_request():
...
+        msg = 'Cannot execute more than one script at the same time.'
+        return msg
+
+    file_name = join(sessiondir, "cleandata.txt")
+    xye = request.form['xye']
+    '''
+    sample input to scilab:
+    xye: 0,1,2,10,20,-30
+    '''
+    command = "clearfun('xinfo');function xinfo(msg),disp(msg),endfunction;"
```
Comment (sunilshetye) [inline SendLog.py]: This is common code. Add it to SCILAB_START above.

File: SendLog.py [hunk 2]
```diff
@@ -1724,6 +1726,110 @@ def run_scilab_func_expr_request():
...
+
+    proc = scifile.instance.proc
+    proc.communicate()
+    remove_scilab_instance(scifile.instance)
+    scifile.instance = None
+
+    valuesfordo_spline = {}
+    '''
+    sample output from scilab:
+    []
```
Comment (sunilshetye) [inline SendLog.py]: is this the correct sample output?

File: SendLog.py [hunk 3]
```diff
@@ -1724,6 +1726,110 @@ def run_scilab_func_expr_request():
...
+    x = request.form['x']
+    y = request.form['y']
+    '''
+    sample input to scilab:
+    N: 1
+    order: 3
+    x: 0
+    y: 1
+    '''
+    command = "clearfun('xinfo');function xinfo(msg),disp(msg),endfunction;"
```
Comment (sunilshetye) [inline SendLog.py]: As mentioned, add to SCILAB_START and remove from here.

File: scifunc-cleandata-do_spline.sci [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+endfunction
+
+function Do_Spline_write(varargin)
+    loop = argn(2);
+    f_temp = varargin(loop);
+    for k = 1:(loop-1)
+            mfprintf(f_temp, '[');
+            variable = varargin(k) // Storing variable one at a time
+            [m,n] = size(variable) // reading the size of variable
+    for y = 1:n // no. of rows in variables
```
Comment (sunilshetye) [inline scifunc-cleandata-do_spline.sci]: indent this block properly. should match previous indentation.

File: webapp/dependencies.js [hunk 5]
```diff
@@ -2151,3 +2151,79 @@ function get_expr_output_for_DefineandSet(head,exx) {
...
+        }
+    });
+    return response;
+}
+
+//For Sigbuilder block
+function getmethod(mtd){
+    var METHOD = "";
+    switch (mtd){
+        case 0:
```
Comment (sunilshetye) [inline webapp/dependencies.js]: Put each case in one line to make it more readable like this:

     case 0: METHOD = "zero order"; break;
     case 1: METHOD = "linear"; break;

File: SendLog.py (hunk 1 above)
Comment (diptighosalkar) [reply SendLog.py]: ok

File: SendLog.py (hunk 2 above)
Comment (diptighosalkar) [reply SendLog.py]: ya , for the above inputs, it is correct sample output

File: SendLog.py (hunk 3 above)
Comment (diptighosalkar) [reply SendLog.py]: ok

File: scifunc-cleandata-do_spline.sci (hunk 4 above)
Comment (diptighosalkar) [reply scifunc-cleandata-do_spline.sci]: ok

File: SendLog.py [hunk 6]
```diff
@@ -1740,6 +1743,108 @@ def run_scilab_func_expr_request():
...
+
+    proc = scifile.instance.proc
+    proc.communicate()
+    remove_scilab_instance(scifile.instance)
+    scifile.instance = None
+
+    valuesforcleandata = []
+    '''
+    sample output from scilab:
+    0 1
```
Comment (sunilshetye) [inline SendLog.py]: once done, please update the sample output also

File: SendLog.py [hunk 7]
```diff
@@ -1740,6 +1743,108 @@ def run_scilab_func_expr_request():
...
+    file_name = join(sessiondir, "do_spline.txt")
+    N = request.form['N']
+    order = request.form['order']
+    x = request.form['x']
+    y = request.form['y']
+    '''
+    sample input to scilab:
+    N: 1
+    order: 3
+    x: 0
```
Comment (sunilshetye) [inline SendLog.py]: update the sample input as per previous sample output

File: SendLog.py (hunk 2 above)
Comment (sunilshetye) [reply SendLog.py]: update this also as per updated input.

**Pull request merged**